test/hex_api_SUITE.erl

```diff
@@ -73,9 +73,25 @@ keys_test(_Config) ->
 
     {ok, {200, _, Key}} = hex_api_key:get(?CONFIG, Name),
 
-    Permissions = [#{<<"domain">> => <<"api">>, <<"resource">> => <<"read">>}],
+    Permissions = [#{domain => <<"api">>, resource => <<"read">>}],
```

Comment:
I suppose binary keys are technically valid, but doesn't jive with the current specs.

Reply:
per #71 we wanted to move forward with binary keys. Since we can't use binary literals then the type we used to have:

```erlang
-type api_permission() :: #{domain := api, resource => read | write}.
```

now would have to be:

```erlang
-type api_permission() :: #{binary() => binary()}.
```

which is not super useful so might as well remove it.

since we lose expressiveness maybe #71 is a mistake, i.e. we should instead standardise on *atom* keys as implementation is supposed to follow the specification and so the key names should be known? I'm now a little bit on the fence.

Reply:
I'd say keep atom keys IMO, the issue I ran into was atom values, not keys. Of course, having some context on why the server rejects binary values such as `<<"api">>`  but accepts atom keys might help here.